### PR TITLE
feat(ras-acc): update otp email template

### PR DESCRIPTION
**Pull request by chickenn00dle**
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

**PR 1 of 4**

This PR updates the OTP signin email with:

- A new footer per RAS-ACC email designs
- Use of primary and secondary color backgrounds
- Some alignment updates

https://github.com/Automattic/newspack-plugin/assets/17905991/1eb55e9a-802b-4a6a-a6fa-e6859682e9d4

Note: Dynamic text color and FAQ sections added in https://github.com/Automattic/newspack-plugin/pull/3033 and https://github.com/Automattic/newspack-plugin/pull/3034

Note: I am only updating the OTP email for now. Once all of the kinks are worked out it should be pretty simple to update the others.

### How to test the changes in this Pull Request:

1. Go to the WP Customizer and set a custom primary and secondary color
2. With these changes checked out, go to Newspack > Engagment > Show Advanced Settings and select the one-time password email under Transactional Email Content section:
Screenshot 2024-04-04 at 15 05 26
3. Trash the post (so that we can force our updated template to be used on next otp sign in attempt)
4. As a logged out user, select the sign in link and attempt to login via email to trigger an otp email
5. Check the otp email and confirm the above points are implemented in the email design

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1205668937699542

### CODE REVIEW

**Comment by chickenn00dle** [comment]
I'm gonna mark this one ready for review despite more work needing to be done to match the designs to hopefully help make review a bit simpler.

I should also note that there is a bug in the newsletters plugin (which it seems the main plugin emails rely on to generate email html) which causes the editor to force buttons to be centered. So our default OTP template will have a correctly left aligned button, but if you go into the post editor for the email template, make any changes, and save then the button will become centered. Similarly, if you send yourself a test email via the post editor, the button will also be centered.

I spent a bunch of time trying to figure out why this is happening, but had no luck as of now. I will tackle this, as well as some of the other email designs in another PR.

File: assets/other-scripts/emails/style.scss
Comment (chickenn00dle) [inline assets/other-scripts/emails/style.scss:8]: Added these styles to resolve an issue where the button block was forced to be centered in the editor. This overrides the behavior in the editor, but actual emails will still see centered buttons for some reason. (Aside from our updated OTP default template)

**Review by adekbadek — CHANGES_REQUESTED** [review]
> Trash the post (so that we can force our updated template to be used on next otp sign in attempt)

1. How will migration work for the sites out in the wild?

2. How will the template be updated in the event of site colors update?

**Comment by chickenn00dle** [comment]
Great questions @adekbadek 

> How will migration work for the sites out in the wild?

I think we want to avoid forcing sites to migrate since many publishers have likely put in some work on updating these emails on their end. My thought was to add an option to reset templates for those that would like to use our new designs. We might also want to add a similar reset link to each email template next to the edit link in the Transactional Email Content section:
Screenshot 2024-04-04 at 15 05 26

That said, I was hoping to focus this PR strictly on the design portion of this and tackle these issues in a follow up PR. Happy to add this in here as well if you prefer though!

> How will the template be updated in the event of site colors update?

This one is a bit tricky for publishers that have spent time customizing their email templates. Again, I'd avoid forcing these publishers to fall back to our designs on color update, and leave it up to them to "reset" via the links mentioned above, OR manually update their custom emails on their end.

I'm definitely open to other suggestions for both of these questions too if you have any thoughts!

**Comment by adekbadek** [comment]
> I think we want to avoid forcing sites to migrate since many publishers have likely put in some work on updating these emails on their end

I agree. But publishers who had not touched the templates should have them updated automatically. 

> > How will the template be updated in the event of site colors update?

> I'd avoid forcing these publishers to fall back to our designs on color update, and leave it up to them to "reset" via the links mentioned above, OR manually update their custom emails on their end.

As a publisher, I'd expect brand colors to be updated in the emails if they are updated for the site, even if email copy has changes.

**Comment by chickenn00dle** [comment]
> I agree. But publishers who had not touched the templates should have them updated automatically.

> As a publisher, I'd expect brand colors to be updated in the emails if they are updated for the site, even if email copy has changes.

Sure! I'm happy to address this in a follow up PR, unless you feel it needs to be implemented here. My thought was to keep this PR and some follow-ups focused on the email redesign and handle the reset/update functionality separately.

File: includes/util.php
Comment (adekbadek) [inline includes/util.php]: Newspack Theme functions should be used ([like here](https://github.com/Automattic/newspack-plugin/blob/36238e96a863723720979520400a9afc65dd0cfa/includes/wizards/class-setup-wizard.php#L332-L333)).

**Review by adekbadek — APPROVED** [review]
Approving pending one small change requested and keeping in mind that the migration strategy should be addressed (as discussed above) before merging the epic branch.

File: includes/util.php
Comment (chickenn00dle) [reply includes/util.php]: Nice! I've fixed this in https://github.com/Automattic/newspack-plugin/pull/3032/commits/cc2f29806f1a07870d51adea496646c0d4e247e8

**Comment by chickenn00dle** [comment]
> keeping in mind that the migration strategy should be addressed

As a heads up @adekbadek, I've got some additional PRs up to tackle:

- Migrating unmodified transactional emails: https://github.com/Automattic/newspack-plugin/pull/3054
- Triggering transactional email updates on theme color change: https://github.com/Automattic/newspack-plugin/pull/3052
- Adding a link to manually reset transactional email templates: https://github.com/Automattic/newspack-plugin/pull/3049

These three PRs should collectively address migration concerns.

**Pull request merged**
